osu.Game.Rulesets.Catch/Skinning/Legacy/LegacyCatchHitExplosion.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Rulesets.Catch.Objects;
+using osu.Game.Rulesets.Catch.UI;
+using osu.Game.Skinning;
+using osuTK;
+
+namespace osu.Game.Rulesets.Catch.Skinning.Legacy
+{
+    public class LegacyCatchHitExplosion : CatchHitExplosion
+    {
+        private readonly Sprite explosion1;
+        private readonly Sprite explosion2;
+
+        /// <inheritdoc cref="Catcher.ALLOWED_CATCH_RANGE"/>
+        private const float catcher_margin = (1 - Catcher.ALLOWED_CATCH_RANGE) / 2;
+
+        [Resolved]
+        protected ISkinSource Skin { get; private set; }
+
+        public LegacyCatchHitExplosion(ISkinSource source)
+        {
+            Anchor = Anchor.CentreLeft;
+            Origin = Anchor.BottomCentre;
+            RelativeSizeAxes = Axes.Both;
+
+            InternalChildren = new Drawable[]
+            {
+                explosion1 = new Sprite
+                {
+                    Origin = Anchor.CentreLeft,
+                    Texture = source.GetTexture("scoreboard-explosion-2"),
+                    Blending = BlendingParameters.Additive,
+                    Rotation = -90,
+                },
+                explosion2 = new Sprite
+                {
+                    Origin = Anchor.CentreLeft,
+                    Texture = source.GetTexture("scoreboard-explosion-2"),
+                    Blending = BlendingParameters.Additive,
+                    Rotation = -90,
+                }
+            };
```

Comment:
It is not actually required to pass `Source` in the ctor. It can be done in BDL.

An example of this can be seen [here](https://github.com/ppy/osu/blob/5e427b43844abbfbe40aa4468eebc106e88a457c/osu.Game.Rulesets.Taiko/Skinning/Legacy/TaikoLegacyPlayfieldBackgroundRight.cs#L21-L42).